extractor/src/extract/extractor.py

```diff
@@ -194,11 +192,19 @@ def split_dataframe(df, analysis):
         pk_col = analysis.primary_key_column.dataframe_column_name()
 
         prev_pk_val = None
-        acc = []
-        for _, row in df.iterrows():
+        acc = defaultdict(list)
+        for row in df:
             if acc and row[pk_col] != prev_pk_val:
-                yield pd.DataFrame(acc, columns=df.columns)
-                acc = []
-            acc.append(row)
+                yield acc
+                acc = defaultdict(list)
+            for key, value in row.items():
```

Comment:
do you think that iterating over each column of each row affects the performance ? maybe it's the same as building a dataframe though ?

Reply:
Yeah, I didn't really find a better way... But as we won't have thousands of columns here, I don't think it's too much of a problem